File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    //these params are in query string of http request
    public class UserParams
    {
        private const int MaxPageSize = 50;
        public int PageNumber { get; set; } = 1;
        private int _pageSize = 10; //our default value
        public int PageSize
        {
            get { return _pageSize; }
            set { _pageSize = (value > MaxPageSize) ? MaxPageSize : value; }
        }

        public string CurrentUserName { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
        public bool OrderIsDescending { get; set; } = true;
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    pageNumber = 1;
    pageSize = 5;
    gender: string;
    minAge = 18;
    maxAge = 99;
    orderBy = 'lastActive';
    orderIsDescending = true;

    constructor(user: User) {
        this.gender = user.gender === 'male' ? 'female' : 'male';
    }
}